starknet_py/abi/v2/parser_transformer.py

```diff
@@ -109,6 +112,17 @@ def type_uint(self, value: List[Token]) -> UintType:
         """
         return UintType(int(value[0]))
 
+    def type_bounded_int(self, value: List[Token]) -> UintType:
+        """
+        BoundedInt Uint type contains information about its ranges. They are present in the value[0] and value[2].
+        """
+        if value[0] != "0":
+            raise ValueError("BoundedInt should start from 0.")
```

Comment:
is this documented somewhere, eg in starknet docs/comments in the code/whatever? I'd like to make sure this is actually true :)

Reply:
It isn't documented, but to prevent unexpected behavior in the code, I added this line. I'm not aware of any other usage of BoundedInt. (It isn't documented)

Reply:
sure, so where did you get this info from? Is this assumption safe?

Reply:
In starknet.py we do not have any data type that is limited from left. So I made this assumption myself. I do not know if in ABI can be occurred BoundedInt with lower bound other than 0 eg. `"type": "core::internal::BoundedInt::<32, 63>"`. If not this raise never be raised if yes, users actually won't be able to use their ABI. I think there is other solution, I can remove this 'if' and 'raise' and believe in users that they will be using values higher than lower bound. But I don't think so such datatype exists.

Reply:
ok, thanks for clarification. I guess we can leave this as is for now, but it would be cool if we could seek a clarification on this somehow